crates/evm/src/tests/sys_tx_tests.rs

```diff
@@ -0,0 +1,165 @@
+
+    let system_account = evm.accounts.get(&SYSTEM_SIGNER, &mut working_set).unwrap();
+    // The system caller balance is unchanged(if exists)/or should be 0
+    assert_eq!(system_account.info.balance, U256::from(0));
+    assert_eq!(system_account.info.nonce, 0);
+
+    // New L1 block №2
+    evm.begin_soft_confirmation_hook(
+        [2u8; 32],
+        2,
+        [3u8; 32],
+        &[10u8; 32],
+        l1_fee_rate,
+        &mut working_set,
+    );
+    {
+        let sender_address = generate_address::<C>("sender");
+        let sequencer_address = generate_address::<C>("sequencer");
+        let context = C::new(sender_address, sequencer_address, 1);
+
+        evm.call(
+            CallMessage {
+                txs: vec![deploy_message],
+            },
+            &context,
+            &mut working_set,
+        )
+        .unwrap();
+    }
+    evm.end_soft_confirmation_hook(&mut working_set);
+    evm.finalize_hook(&[99u8; 32].into(), &mut working_set.accessory_state());
+
+    let system_account = evm.accounts.get(&SYSTEM_SIGNER, &mut working_set).unwrap();
+    // The system caller balance is unchanged(if exists)/or should be 0
+    assert_eq!(system_account.info.balance, U256::from(0));
+    assert_eq!(system_account.info.nonce, 0);
+
+    let receipts: Vec<_> = evm
+        .receipts
+        .iter(&mut working_set.accessory_state())
+        .collect();
+    assert_eq!(receipts.len(), 4); // 2 from #1 L1 block and 2 from #2 block
+    let receipts = receipts[2..].to_vec();
+
+    assert_eq!(receipts,
+        [
+            Receipt { // L1BlockHashList::setBlockInfo(U256, U256)
+                receipt: reth_primitives::Receipt {
+                    tx_type: reth_primitives::TxType::Eip1559,
+                    success: true,
+                    cumulative_gas_used: 75665,
+                    logs: vec![
+                        Log {
+                            address: L1BlockHashList::address(),
+                            topics: vec![b256!("32eff959e2e8d1609edc4b39ccf75900aa6c1da5719f8432752963fdf008234f")],
+                            data: Bytes::from_static(&hex!("000000000000000000000000000000000000000000000000000000000000000302020202020202020202020202020202020202020202020202020202020202020303030303030303030303030303030303030303030303030303030303030303")),
+                        }
+                    ]
+                },
+                gas_used: 75665,
+                log_index_start: 0,
+                diff_size: 412,
+                error: None
+            },
+            Receipt {
+                receipt: reth_primitives::Receipt {
+                    tx_type: reth_primitives::TxType::Eip1559,
+                    success: true,
+                    cumulative_gas_used: 189900,
+                    logs: vec![]
+                },
+                gas_used: 114235,
+                log_index_start: 1,
```

Comment:
this has no nonce but log index start is 1

are we sure about this?

Reply:
Yes because they are in the same L2 block.